### PR TITLE
Docker compose tests

### PR DESCRIPTION
**Pull request by hellais**
This PR achieves the following:
* Switch to s3 for the hosting of the debian builds
* Port the workflow for building packages over to github actions
* Create a docker-compose based system for running tests on developers laptop and on the CI

This is based on top of https://github.com/ooni/api/pull/240

### CODE REVIEW

File: newapi/api.conf.example [hunk 1]
```diff
@@ -23,11 +23,11 @@ TOR_TARGETS_CONFFILE = "/etc/ooni/tor_targets.json"
 # Registration email delivery
 MAIL_SERVER = "CHANGEME"
 MAIL_PORT = 465
-MAIL_USE_SSL = True
+MAIL_USE_SSL = False
 MAIL_USERNAME = "CHANGEME"
 MAIL_PASSWORD = "CHANGEME"
 MAIL_SOURCE_ADDRESS = "contact@ooni.org"
-LOGIN_BASE_URL = "https://CHANGEME/login"
+LOGIN_BASE_URL = "https://api.ooni.io/api/v1/user_login"
```
Comment (FedericoCeratto) [inline newapi/api.conf.example:30]: This is an example for 3rd parties: we should not hardcode parameters related to the ooni infra.

Comment (hellais) [reply newapi/api.conf.example:30]: It's a dependency for the tests to run properly. Should I split out the config file used for testing from the config file used as example?

File: .github/workflows/mypy.yml [hunk 2]
```diff
@@ -13,14 +13,16 @@ jobs:
       - name: Setup APT
         run: |
           apt-get update
-          apt-get install --no-install-recommends -y ca-certificates
+          apt-get install --no-install-recommends -y ca-certificates gnupg
```
Comment (FedericoCeratto) [inline .github/workflows/mypy.yml:16]: Do we need gnupg here?

Comment (hellais) [reply .github/workflows/mypy.yml:16]: Don't we need it to install the GPG key of the deb-ci repo?

Comment (FedericoCeratto) [reply .github/workflows/mypy.yml:16]: no, that's done by apt itself

File: newapi/api.conf.example (hunk 1 above)
Comment (FedericoCeratto) [reply newapi/api.conf.example]: Yes: the example for 3rd party users should have "CHANGEME"s to indicate  what needs customization. The files for local integ test and github CI have specific values for those envs.

**Review by FedericoCeratto — CHANGES_REQUESTED** [review]
The CI run in github is going to become very complex especially once we start running the fastpath in it. We need to ensure that tmate works reliably in order to be able to debug and maintain it.

**Comment by hellais** [comment]
Added support for manually triggering tmate runs

**Review by FedericoCeratto — APPROVED** [review]

**Pull request merged**